entry/src/entry.rs

```diff
@@ -137,7 +137,16 @@ pub struct Entry {
     /// The number of hashes since the previous Entry ID.
     pub num_hashes: u64,
 
-    /// The SHA-256 hash `num_hashes` after the previous Entry ID.
+    /// When `transactions` is empty, this is a SHA-256 hash applied `num_hashes` times to the
+    /// `hash` field of the previous `Entry`.
+    ///
+    /// When `transactions` is non-empty, first SHA-256 hash is applied `num_hashes - 1` times to
+    /// the `hash` field of the previous `Entry`. Resulting hash is is concatenated with the root
+    /// of a [`MerkleTree`] containing all all signatures in all `transactions`, and hashed with
+    /// SHA-256 once again.
+    ///
+    /// [`next_hash()`] implements this computation.
+    /// [`hash_transactions`] constructs the root of a [`MerkleTree`] from the signatures.
```

Comment:
What about just linking to `next_hash()` and handling documentation there?

So something here like: "The output of running `next_hash()` on the `hash` field of the previous `Entry` with this `Entry`'s `num_hashes` and `transactions` values"
I'm not happy with that wording, but hopefully that gives the idea.

That way, in the unlikely event that we do change the poh hashing mechanism in `next_hash`, we don't have to remember to update docs in another spot.

Reply:
One problem that I faced while trying to understand Solana codebase was the absence of higher level descriptions.
I can read the code, but it is much slower, compared to reading a higher level description of the intent of a certain subsystem or abstraction.
I have not seen a problem of a stale comment yet.

As far as I understand, entries, transactions, and PoH is pretty fundamental to Solana.
So, it is unlikely their relationship changes.
I see this description as a description of the Solana protocol.
Any further changes here will require a SIMD.  And anybody making those, ideally, should also update the doc-comments.
I think, the risk of it getting the comment out of sync with the code is low and acceptable, compared to the convenience of having a fuller description of an entry in one place.